src/NServiceBus.Core/Routing/AutomaticSubscriptions/AutoSubscribe.cs

```diff
@@ -32,8 +32,7 @@ protected internal override void Setup(FeatureConfigurationContext context)
             }
 
             var conventions = context.Settings.Get<Conventions>();
-            var transportInfrastructure = context.Settings.Get<TransportInfrastructure>();
-            var requireExplicitRouting = transportInfrastructure.OutboundRoutingPolicy.Publishes == OutboundRoutingType.Unicast;
+            var requireExplicitRouting = context.Transport.TransportInfrastructure.OutboundRoutingPolicy.Publishes == OutboundRoutingType.Unicast;
```

Comment:
Minor stuff: omitting `Transport` in `TransportInfrastructure` prop would make it easier to read.